superset/key_value/commands/get.py

```diff
@@ -20,26 +20,24 @@
 
 from flask import current_app as app
 from flask_appbuilder.models.sqla import Model
-from flask_appbuilder.security.sqla.models import User
 from sqlalchemy.exc import SQLAlchemyError
 
 from superset.commands.base import BaseCommand
+from superset.key_value.commands.args import Args
 from superset.key_value.commands.exceptions import KeyValueGetFailedError
 
 logger = logging.getLogger(__name__)
 
 
 class GetKeyValueCommand(BaseCommand, ABC):
-    def __init__(self, actor: User, resource_id: int, key: str):
-        self._actor = actor
-        self._resource_id = resource_id
-        self._key = key
+    def __init__(self, args: Args):
+        self._args = args
 
     def run(self) -> Model:
         try:
             config = app.config["FILTER_STATE_CACHE_CONFIG"]
-            refresh_timeout = config.get("REFRESH_TIMEOUT_ON_RETRIEVAL")
-            return self.get(self._resource_id, self._key, refresh_timeout)
+            self._args["refresh_timeout"] = config.get("REFRESH_TIMEOUT_ON_RETRIEVAL")
+            return self.get(self._args)
```

Comment:
Bycatch - we probably can't be referencing `app.config["FILTER_STATE_CACHE_CONFIG"]` on line 38 here. Maybe we need to pass the cache config key to `self._args` as it will be different for each kv implementation?

Reply:
Oh, you're right! I moved the logic to the implementations themselves.